### PR TITLE
ZATS-79: Modernize ZATS for ZK 10 Jakarta and Jetty 12 compatibility

### PR DESCRIPTION
**Pull request by szediwy**
### Purpose
This PR modernizes the ZATS core library to fully support ZK 10 Jakarta and ensures compatibility with Jetty 12 and modern Java environments, resolving **ZATS-79**.

### Summary of Improvements
- **Jakarta & Infrastructure Alignment**: Updated core dependencies and build configurations to support the Jakarta namespace and modern Java Multi-Release JAR loading requirements (e.g., `IOIterator` in `commons-io`).
- **Improved Test Stability**: Refactored environment initialization to automatically handle existing emulators, eliminating cascading "already started up" failures and ensuring reliable test isolation.
- **Enhanced Platform Compatibility**: Resolved path resolution issues (canonical vs absolute) that previously impacted static resource serving on macOS and other modern server environments.
- **Test Suite Modernization**: Comprehensive update of the existing test suite to align with modern ZK component behaviors, visibility logic, and event processing lifecycles.

### Compliance
By contributing this code, I certify that I created the content and agree to surrender the copyright of the contributed code to Potix Corporation, as per the ZATS contribution guidelines.

### CODE REVIEW

**Comment by jeanher** [comment]
Hello Szediwy, Thank you for creating the pull request and for helping improve Jakarta support.

Before we proceed, please make sure you have read and agree to the [contribution guidelines](https://github.com/zkoss/zats/blob/master/contributing.md), as they include information about contribution and licensing terms. Once confirmed, we will review your submission and get back to you.

If your contribution is merged, we would also be happy to acknowledge you in the release notes. Please let us know how you would like to be credited. Example: Huge thanks to [NAME] for contributing the [Feature]. 

Thank you for helping to make ZK better.

**Comment by szediwy** [comment]
I have read and agree to the [ZATS contribution guidelines](https://github.com/zkoss/zats/blob/master/contributing.md). By contributing this code, I certify that I created the content and agree to surrender the copyright of the contributed code to Potix Corporation.

Please credit me as **Michael Szediwy** for contributing the **Modernization for ZK 10 Jakarta and Jetty 12 support**.

**Comment by szediwy** [comment]
I have also previously contributed to ZK with this GitHub account (ZK Tracker login: **mis**).

**Comment by jeanher** [comment]
Thank you for the confirmation. I can see that you’ve been part of the community for 15 years — wonderful to see such long-time support!

File: zats-mimic/src/main/java/org/zkoss/zats/mimic/impl/emulator/JettyEmulator.java [hunk 1]
```diff
@@ -118,20 +109,49 @@ public void doHandle(String target, Request baseRequest, HttpServletRequest requ
...
 			final BlockingQueue<Object> queue = new ArrayBlockingQueue<Object>(1, true);
 
-			contextHandler.addEventListener(new Listener() {
+			contextHandler.addEventListener(new LifeCycle.Listener() {
 				@Override
 				public void lifeCycleStarted(LifeCycle event) {
+                    try {
+                        System.err.println("DEBUG: ClassLoader for WebAppContext: " + contextHandler.getClassLoader());
+                        Class<?> clazz = contextHandler.getClassLoader().loadClass("org.apache.commons.io.function.IOIterator");
+                        System.err.println("DEBUG: Loaded IOIterator from: " + clazz.getProtectionDomain().getCodeSource().getLocation());
```
Comment (DevChu) [inline zats-mimic/src/main/java/org/zkoss/zats/mimic/impl/emulator/JettyEmulator.java]: Recommend not to use System.err.println here. It is for debugging.
Please use logger.debug

File: zats-mimic/src/main/java/org/zkoss/zats/mimic/impl/EmulatorClient.java [hunk 2]
```diff
@@ -298,6 +299,13 @@ private String getCombinedEventString(String desktopId, List<AbstractUploadAgent
 		return sb.toString();
 	}
 
+	private String encode(String value) {
+		MultiMap<String> mm = new MultiMap<>();
+		mm.add("v", value);
```
Comment (DevChu) [inline zats-mimic/src/main/java/org/zkoss/zats/mimic/impl/EmulatorClient.java]: Wondering why need to add "v" -> encode -> then remove?

File: zats-mimic/src/test/java/org/zkoss/zats/testcase/BasicAgentTest.java [hunk 3]
```diff
@@ -5,35 +5,25 @@
 	Description:
 		
 	History:
-		2012/3/22 Created by dennis
```
Comment (DevChu) [inline zats-mimic/src/test/java/org/zkoss/zats/testcase/BasicAgentTest.java:8]: It seems that no need to change this.

File: zats-mimic/src/test/resources/web/basic/click-all.zul [hunk 4]
```diff
@@ -1,151 +1,87 @@
...
\ No newline at end of file
+	]]></zscript>
+	<vbox id="comps">
+		<label id="history" value="" />
+		<a id="a" label="a" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)"/>
+		<button id="button" label="button" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)"/>
+		<captcha id="captcha" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)"/>
+		<fileupload id="fileupload" label="fileupload" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)"/>
+		<div id="html" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)">html</div>
+		<include id="include" src="~./basic/hello.zul" onClick="show(event)" onDoubleClick="show(event)" onRightClick="show(event)" onMouseOver="show(event)" onMouseOut="show(event)"/>
```
Comment (DevChu) [inline zats-mimic/src/test/resources/web/basic/click-all.zul:17]: There is no hello.zul under basic folder.
might missing something

**Review by DevChu — COMMENTED** [review]
Thanks for the update — this PR looks much better after the follow-up commits.
I can see the previous review points were addressed: debug prints were replaced with logger calls, the speculative event payload changes were cleaned up, width/height handling for maximize/minimize was restored, and the dropped tests/components were added back. The overall Jetty 12 / Jakarta EE10 alignment in zats-mimic/pom.xml also looks consistent with the PR goal.

I only have one substantive concern left: DefaultZatsEnvironment.init() changed from throwing "already started up" to implicitly calling destroy(). That is a behavior change, not just a test-stability fix. Could we either add a dedicated test for the new semantics or document that re-init is now intentionally supported?

One minor suggestion: since encode() was simplified to URLEncoder.encode(...), it would be good to add one focused test covering special characters (space, +, %, &, Unicode), just to make sure the AU request payload stays fully compatible.

**Comment by DevChu** [comment]
Thanks for the follow-up commit.
The added tests cover the remaining concerns I had, and the PR now looks good to me. LGTM.

**Comment by DevChu** [comment]
@szediwy 
Thank you again for your contribution and for helping fix this bug.

We reviewed and merged your pull request earlier.
The reason it was not released immediately is that we wanted to upgrade the ZK version to 10.2.1 first, so the fix is now included in the latest release:

10.2.1-jakarta-FL-2026040819

We really appreciate your support and contribution to the project.
